share/ci/generate_reduced_matrix.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+set -e
+set -o pipefail
+
+# generate a job reduced matrix based on the environment variable lists (space separated)
```

Comment:
`generate a job reduced` - probably, some preposition or another word is missing after `job`